packages/superset-ui-plugin-chart-table/src/Table.tsx

```diff
@@ -0,0 +1,221 @@
+    TableState
+  > = (props: InternalTableProps, state: TableState) => {
+    const { filters } = props;
+    const { selectedCells, filters: prevFilters } = state;
+    if (prevFilters !== filters) {
+      const newSelectedCells = new Set(Array.from(selectedCells));
+      Object.keys(filters).forEach(key => {
+        filters[key].forEach(value => {
+          newSelectedCells.add(
+            getCellHash({
+              key,
+              value,
+            }),
+          );
+        });
+      });
+
+      return {
+        ...state,
+        filters,
+        selectedCells: newSelectedCells,
+      };
+    }
+
+    return state;
+  };
+
+  handleCellSelected = (cell: Cell) => () => {
+    const { selectedCells } = this.state;
+    const { tableFilter, onRemoveFilter, onAddFilter } = this.props;
+
+    if (!tableFilter) {
+      return;
+    }
+    const newSelectedCells = new Set(Array.from(selectedCells));
+    const cellHash = getCellHash(cell);
+    if (newSelectedCells.has(cellHash)) {
+      newSelectedCells.delete(cellHash);
+      onRemoveFilter(cell.key, [cell.value as number]);
+    } else {
+      newSelectedCells.add(cellHash);
+      onAddFilter(cell.key, [cell.value as number]);
+    }
+    this.setState({
+      selectedCells: newSelectedCells,
+    });
+  };
+
+  isSelected = (cell: Cell) => {
+    const { selectedCells } = this.state;
+
+    return selectedCells.has(getCellHash(cell));
+  };
+
+  handleSearch = (value: string) => {
+    const { searchKeyword } = this.state;
+    const { data } = this.props;
+    if (searchKeyword !== value) {
+      const filteredRows = data.filter(row => {
+        const content = Object.keys(row.data)
+          .map(key => row.data[key])
+          .join('|')
+          .toLowerCase();
+
+        return content.indexOf(value.toLowerCase()) >= 0;
+      });
+      this.setState({
+        filteredRows,
+        searchKeyword: value,
+      });
+    }
+  };
+
+  render() {
+    const {
+      data,
+      columns,
+      alignPositiveNegative,
+      colorPositiveNegative,
+      height,
+      tableFilter,
+      styles,
+      includeSearch,
+    } = this.props;
+
+    const { filteredRows, searchKeyword } = this.state;
+
+    const renderers: Renderers = {};
+
+    const dataToRender = searchKeyword === '' ? data : filteredRows;
+    const columnMetadata: ColumnMetadata = {};
+
+    columns.forEach(column => {
+      renderers[column.key] = getRenderer({
+        alignPositiveNegative,
+        colorPositiveNegative,
+        column,
+        enableFilter: tableFilter,
+        handleCellSelected: this.handleCellSelected,
+        isSelected: this.isSelected,
+      });
+      if (column.type === 'metric') {
+        columnMetadata[column.key] = {
+          rightAlign: 1,
+        };
+      }
+    });
+
+    return (
+      <>
+        {includeSearch && (
+          <div {...css(styles.searchBar)}>
+            <div {...css(styles.searchBox)}>
+              <Input
```

Comment:
Does the `DataTable` now include built-in search box?

Reply:
There is no built-in search box for `<DataTable/>`, and in any case, users need to implement the search logic by themself, the example here is https://github.com/airbnb/lunar/blob/master/packages/core/src/components/DataTable.story.tsx#L121.  

However, using it will prevent us from showing the message like `show ** out of ** row`, and the benefit of using `filterData` is to keep the status of `expended rows`, which we don't really use here.  

So I think we should just use the logic written here.